File: voice-note-server/src/main/kotlin/com/suikouji/server/llm/dto/TransactionParseRequest.kt
package com.suikouji.server.llm.dto

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

/** Client request to parse natural language into structured transaction data. */
data class TransactionParseRequest(
    @field:NotBlank
    @field:Size(max = 2000, message = "Text must be at most 2000 characters")
    val text: String,

    @field:Valid
    val context: ParseContext? = null
)

/** Optional context to improve parsing accuracy. */
data class ParseContext(
    @field:Size(max = 50, message = "recentCategories cannot exceed 50 items")
    val recentCategories: List<@Size(max = 100) String>? = null,

    @field:Size(max = 50, message = "customCategories cannot exceed 50 items")
    val customCategories: List<@Size(max = 100) String>? = null,

    @field:Size(max = 20, message = "accounts cannot exceed 20 items")
    val accounts: List<@Size(max = 100) String>? = null
)

File: voice-note-client/lib/core/network/dto/transaction_parse_request.dart
/// Request body for POST /api/v1/llm/parse-transaction.
class TransactionParseRequest {
  final String text;
  final ParseContext? context;

  const TransactionParseRequest({
    required this.text,
    this.context,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{'text': text};
    if (context != null) json['context'] = context!.toJson();
    return json;
  }
}

/// Optional context to improve LLM parsing accuracy.
class ParseContext {
  final List<String>? recentCategories;
  final List<String>? customCategories;
  final List<String>? accounts;

  const ParseContext({
    this.recentCategories,
    this.customCategories,
    this.accounts,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{};
    if (recentCategories != null) json['recentCategories'] = recentCategories;
    if (customCategories != null) json['customCategories'] = customCategories;
    if (accounts != null) json['accounts'] = accounts;
    return json;
  }
}
